myserver.cpp

```diff
@@ -0,0 +1,553 @@
+        document.setObject(GetFriendList);
+        bytes = document.toJson( QJsonDocument::Indented );
+        socketWrite(bytes);
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::AddFriend(const std::vector <std::string> &msg){
+    iUserId = std::stoi(msg.at(1));
+    iFriendId = std::stoi(msg.at(2));
+    int iRtCode = IDataBase::AddFriend(iUserId, iFriendId);
+    switch(iRtCode){
+    case SUCCESS:
+        socketWrite("SUCCESS");
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case NO_SUCH_FRIEND:
+        socketWrite("NO_SUCH_FRIEND");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::DeleteFriend(const std::vector <std::string> &msg){
+    iUserId = std::stoi(msg.at(1));
+    iFriendId = std::stoi(msg.at(2));
+    int iRtCode = IDataBase::DeleteFriend(iUserId, iFriendId);
+    switch(iRtCode){
+    case SUCCESS:
+        socketWrite("SUCCESS");
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case NO_SUCH_FRIEND:
+        socketWrite("NO_SUCH_FRIEND");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::GetFriendRequestsList(const std::vector <std::string> &msg){
+    std::vector<std::string> vsFrReqList;
+    int iRtCode = IDataBase::GetFriendRequestsList(iId, vsFrReqList);
+    QJsonObject GetFriendRequestsList;
+    std::string sRqList;
+    switch(iRtCode){
+    case SUCCESS:
+        for(int i = 0; i <= vsFrReqList.size(); i++){
+             sRqList += vsFrReqList[i] + '|';
+        }
+        GetFriendRequestsList.insert("RequestType", "GetFriendRequestsList");
+        GetFriendRequestsList.insert("SUCCESS", sRqList.data());
+        document.setObject(GetFriendRequestsList);
+        bytes = document.toJson( QJsonDocument::Indented );
+        socketWrite(bytes);
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::AddFriendRequest(const std::vector <std::string> &msg){
+    iUserId = std::stoi(msg.at(1));
+    iFrReqId = std::stoi(msg.at(2));
+    int iRtCode = IDataBase::AddFriendRequest(iUserId, iFrReqId);
+    switch(iRtCode){
+    case SUCCESS:
+        socketWrite("SUCCESS");
+        break;
+    case NO_SUCH_USER:
+        socketWrite("NO_SUCH_USER");
+        break;
+    case NO_SUCH_FRIEND:
+        socketWrite("NO_SUCH_FRIEND");
+        break;
+    case UNEXPECTED_ERROR:
+        socketWrite("UNEXPECTED_ERROR");
+        break;
+    }
+}
+void CMyServer::DeleteFriendRequest(const std::vector <std::string> &msg){
+    iUserId = std::stoi(msg.at(1));
```

Comment:
Same here, consider adding checks for valid data

Reply:
Fixed